wercker.yml

```diff
@@ -109,10 +122,90 @@ push:
         username: $OCIRUSERNAME
         password: $OCIRPASSWORD
 
+
+e2e-test:
+  base-path: "/go/src/github.com/oracle/oci-cloud-controller-manager"
+  box:
+    id: iad.ocir.io/oracle/oci-cloud-controller-manager-ci-e2e:1.0.0
+    registry: https://iad.ocir.io/v2
+    username: $OCIRUSERNAME
+    password: $OCIRPASSWORD
+  steps:
+
+    - script:
+      name: set ENV vars
+      code: |
+        export VERSION=$(cat VERSION.txt)
+        echo "${VERSION}"
+
+    - script:
+      name: deploy latest ccm
+      code: make upgrade
+        
+    - script:
+      name: e2e default tests
+      code: make e2e
+    
+    - script:
+      name: rollback original ccm
+      code: make rollback
+
+
+push-canary:
+  base-path: "/go/src/github.com/oracle/oci-cloud-controller-manager"
+  box:
+    id: iad.ocir.io/oracle/oci-cloud-controller-manager-ci-e2e:1.0.1
+    registry: https://iad.ocir.io/v2
+    username: $OCIRUSERNAME
+    password: $OCIRPASSWORD
+  steps:
+    - script:
+      name: set ENV vars
+      code: |
+        export VERSION=$(cat dist/VERSION.txt)
+        echo "Pushing test version ${VERSION}"
+
+    - internal/docker-push:
+      repository: iad.ocir.io/oracle/oci-cloud-controller-manager-canary
+      tag: $VERSION
+      working-dir: /go/src/github.com/oracle/oci-cloud-controller-manager/ 
+      entrypoint: make canary
+      registry: https://iad.ocir.io/v2
+      username: $OCIRUSERNAME
+      password: $OCIRPASSWORD
+
+
+validate-canary:
+  box:
+    id: iad.ocir.io/oracle/oci-cloud-controller-manager-ci-e2e:1.0.1
+    registry: https://iad.ocir.io/v2
+    username: $OCIRUSERNAME
+    password: $OCIRPASSWORD
+  steps:
+    - script:
+      name: set ENV vars
+      code: |
+        export VERSION=$(cat VERSION.txt)
+        echo "Canary image version ${VERSION}"
+
+    - script:
+      name: deploy latest ccm
+      code: make upgrade
+        
+    - script:
+      name: validate canary tests
+      code: make validate-canary
+    
+    - script:
+      name: rollback original ccm
+      code: make rollback
+
```

Comment:
Same here.  Would the deploy/test step failing mean we don't roll back to the old version of the ccm?

Reply:
Yes you are right there! I was meaning to ask Danny to see if Wercker had any way to always run clean-up type tasks. If not I guess the solution will have to wrap up these 3 steps into one step with some error handling. A shame thought cos' I thought this made it nice and clear what was going on...